akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -605,6 +605,71 @@ trait FlowOps[+Out, +Mat] {
   def fold[T](zero: T)(f: (T, Out) ⇒ T): Repr[T, Mat] = andThen(Fold(zero, f.asInstanceOf[(Any, Any) ⇒ Any]))
 
   /**
+   * Intersperses stream with provided element, similar to how [[scala.collection.immutable.List.mkString]]
+   * injects a separator between a List's elements.
+   *
+   * Additionally can inject start and end marker elements to stream.
+   *
+   * Examples:
+   *
+   * {{{
+   * val nums = Source(List(1,2,3)).map(_.toString)
+   * nums.intersperse(",")            //   1 , 2 , 3
+   * nums.intersperse("[", ",", "]")  // [ 1 , 2 , 3 ]
+   * }}}
+   *
+   * In case you want to only prepend or only append an element (yet still use the `intercept` feature
+   * to inject a separator between elements, you may want to use the following pattern instead of the 3-argument
+   * version of intersperse (See [[Source.concat]] for semantics details):
+   *
+   * {{{
+   * Source.single(">> ") ++ Source(List("1", "2", "3")).intersperse(",")
+   * Source(List("1", "2", "3")).intersperse(",") ++ Source.single("END")
+   * }}}
+   *
+   * '''Emits when''' upstream emits (or before with the `start` element if provided)
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def intersperse[T >: Out](start: T, inject: T, end: T): Repr[T, Mat] = {
+    ReactiveStreamsCompliance.requireNonNullElement(start)
```

Comment:
> Consistency is dog :)

There's different flavours of dog :-)
The current state is consistent in the sense of "as soon as possible" - in the iterator we only can fail when looking into it, here we can fail right away. For dev time as-soon-as-possible is usually the best expirience IMO, you know where the null was passed in. Once it's flowing you get "well, some element in that pipeline was null, let's see hm hm..." instead of seeing it right away in the stacktrace – "oh, single was given a null".

I'd prefer the ASAP failing, but if you have strong feelings towards "always the same way - i.e. once running" I can adjust.